r2-netty/src/test/java/com/linkedin/r2/transport/http/client/TestHttpNettyClient.java

```diff
@@ -261,7 +261,7 @@ public void testMaxHeaderSize() throws InterruptedException, IOException, Timeou
   {
     testHeaderSize(TEST_MAX_HEADER_SIZE - 1, RESPONSE_OK);
 
-    testHeaderSize(TEST_MAX_HEADER_SIZE, RESPONSE_OK);
+    testHeaderSize(TEST_MAX_HEADER_SIZE, TOO_LARGE);
```

Comment:
@vivdesh Hi, could you explain why we need to change this test?

Reply:
Mostly strict parsing of the http header causing this test to fail. https://github.com/netty/netty/pull/10058.
Please let me know where we can change the code according to this?

Reply:
@vivdesh Thanks for the additional information. If the original test failed because our header generator doesn't follow strictly the HTTP RFC, we probably better fix it.

Reply:
Actually, there is already a fix from https://github.com/linkedin/rest.li/pull/595/files#diff-bd070627be0acf661a65543c18644309364d2aaf26967596cfdce517c07a6a12R232 by @tjni 
Looking more, this change appears to be a subset of https://github.com/linkedin/rest.li/pull/595 which also introducing a newer version of Netty. It makes sense to me to review and merge that commit instead.